examples/compute/main.c

```diff
@@ -32,7 +52,13 @@ int main(
 
     uint32_t numbers_length = size / sizeof(uint32_t);
 
-    WGPUAdapterId adapter = wgpu_request_adapter(NULL);
+    WGPUAdapterId adapter = { 0 };
+    wgpu_request_adapter_async(
+        NULL,
+        request_adapter_callback,
+        (void *) &adapter
+    );
+
     WGPUDeviceId device = wgpu_adapter_request_device(adapter, NULL);
```

Comment:
so this is incorrect now since we need to wait for the callback?

Reply:
Currently the examples will still work because `request_adapter_callback` is called before `wgpu_adapter_request_device`. They examples will break once we have an event loop available though.

I guess we'll have a similar issue with buffer mapping callbacks once an event loop is available. i.e. we can't use `force_wait` in `wgpu_device_poll` to wait for our callback to be fired inline, and the implementation of `wgpu_device_poll` will be moved into the event loop anyway.

Alternatively, since we don't have a event loop yet, we could move the rest of `main` into the callback to wait for it that way.